test-app/src/main/java/org/readium/r2/testapp/utils/UserError.kt

```diff
@@ -6,68 +6,84 @@
 
 package org.readium.r2.testapp.utils
 
+import android.app.Activity
 import android.content.Context
 import androidx.annotation.PluralsRes
 import androidx.annotation.StringRes
+import androidx.appcompat.app.AlertDialog
+import com.google.android.material.snackbar.Snackbar
 import java.text.DateFormat
 import java.util.Date
 import org.joda.time.DateTime
+import org.readium.r2.shared.util.Error
+import org.readium.r2.shared.util.toDebugDescription
+import org.readium.r2.testapp.R
+import org.readium.r2.testapp.utils.extensions.createShareIntent
+import timber.log.Timber
 
 /**
  * An error that can be presented to the user using a localized message.
  */
 class UserError private constructor(
     val content: Content,
-    val cause: UserError?
+    val cause: Error?
```

Comment:
I'd kept this as it was because that allowed to build hierarchical user messages. I'm not sure that's a good idea though. Do you think so too?

Reply:
I don't think we will ever use this in the Test App. But that's an easy fix whenever it's needed: make `UserError` implement `Error`.

Reply:
I tried it, that's awkward: `UserError` would need to implement two messages: one for debug without accessing context, and one for user. Never mind.